cql3/restrictions/statement_restrictions.cc

```diff
@@ -1340,5 +1340,75 @@ const single_column_restrictions::restrictions_map& statement_restrictions::get_
     return single_restrictions->restrictions();
 }
 
+void statement_restrictions::prepare_indexed(const schema& idx_tbl_schema, bool is_local) {
+    if (is_local || !_partition_range_is_simple) {
+        return;
+    }
+    // This assumes that a base-table partition column cannot be the indexed column:
```

Comment:
Is the assumption validated?

I think that if we have a compound partition key pk1, pk2 then creating an index on either pk1 or pk2 is allowed.

Reply:
Hm, the assumption is, in fact, invalid. #3404

I'll write some tests exercising this and fix the code.

Reply:
Ok. It's okay to move with this independently, as long as we don't behave differently from the status quo.

Reply:
Actually, this code will never be called if the index is on a partition column, because `statement_restrictions::process_partition_key_restrictions()` turns on indexing only if PK restrictions require filtering.  If we are here, it means PK is fully restricted by equalities (`_partition_range_is_simple`), which doesn't require filtering.  It also means an index is being used.  Therefore, the index cannot be on a partition column.

Reply:
I'll replace the comment with an explanation confirming that the assumption is correct.